docs/concepts/controllers.md

```diff
@@ -0,0 +1,110 @@
+
+It watches for changes in the status of the `Statefulset`s associated with the `Etcd` resources.
+Even though the `Etcd` resource owns the `Statefulset`, it is not necessary that the *etcd controller* reconciles whenever there are changes in the statuses of the objects that the `Etcd` resource owns.
+
+Status fields of the `Etcd` resource which correspond to the `StatefulSet` like `CurrentReplicas`, `ReadyReplicas`, `Replicas` and `Ready` are updated to reflect those of the `StatefulSet` by the controller. Cluster membership (`EtcdMemberStatus`) and `Conditions` are updated as follows:
+
+- Cluster Membership: The controller updates the information about etcd cluster membership like `Role`, `Status`, `Reason`, `LastTransitionTime` and identifying information like the `Name` and `ID`. For the `Status` field, the member is checked for the *Ready* condition, where the member can be in `Ready`, `NotReady` and `Unknown` statuses.
+
+- Condition: The controller updates the `Conditions` field which holds the latest information of the `Etcd`'s state. The condition checks that are performed are `AllMembersCheck`, `ReadyCheck` and `BackupReadyCheck`. The first two of these checks make use of the `EtcdMemberStatus` to update `Conditions` with information about the members, and the last corresponds to the status of the backup.
+
+To reflect changes that occur in the `Statefulset` status in the `Etcd` resource, the *custodian controller* keeps a watch on the `Statefulset`.
+
+The *custodian controller* reconciles periodically, which can be set through the `--custodian-sync-period` CLI flag (default being 30 seconds). It also reconciles whenever there are changes to the `Statefulset` status.
+
+The *custodian controller* is essential to the functioning of etcd-druid, thus the minimum number of worker threads is 1, (default being 3).
+
+## Compaction Controller
+
+The *compaction controller* deploys the snapshot compaction job whenever required.
+The controller watches the number of events accumulated as part of delta snapshots in the etcd cluster's backups, and triggers a snapshot compaction when the number of delta events crosses the set threshold, which is configurable through the `--etcd-events-threshold` CLI flag (1M events by default).
+
+The controller watches for changes in *snapshot* `Leases` associated with `Etcd` resources.
+It checks the full and delta snapshot `Leases` and calculates the difference in events between the latest delta snapshot and the previous full snapshot, and initiates the compaction job if the event threshold is crossed.
+
+The number of worker threads for the *compaction controller* needs to be greater than or equal to 0 (default 3).
+This is unlike other controllers who need atleast one worker thread for the proper functioning of etcd-druid.
+This is because snapshot compaction is not a core functionality for the etcd clusters to be deployed.
```

Comment:
same as above

Reply:
Addressed in [d69557c](https://github.com/gardener/etcd-druid/pull/722/commits/d69557cbe510fbaa12e205498ddb02f24d09c697)